front-end/virtual_pet/lib/pages/pomodoro_page.dart

```diff
@@ -1,30 +1,129 @@
+// ignore_for_file: sized_box_for_whitespace
+
 import 'package:flutter/material.dart';
+import 'minutes.dart';
+import 'dart:async';
 import 'package:virtual_pet/widgets/app_bar.dart';
 
 /// pomodoro page (pomodoro timer)
 class PomodoroPage extends StatefulWidget {
   const PomodoroPage({super.key});
-
   @override
   PomodoroState createState() => PomodoroState();
 }
 
 /// the state of pomodoro page (timer)
 class PomodoroState extends State<PomodoroPage> {
   static const String routeName = "/PomodoroPage";
+  int timeLeft = 0;
+  int lastValue = 0;
+  bool inProgress = false;
+  bool firstTime = true;
+
+  @override
+  void initState() {
+    super.initState();
+  }
+
+  void _startCountDown() {
+    setState(() {
+      inProgress = true;
+      firstTime = false;
+      lastValue = timeLeft;
+    });
+    Timer.periodic(const Duration(seconds: 1), (timer) {
+      if (timeLeft > 0) {
+        setState(() {
+          timeLeft--;
+        });
+      } else {
+        setState(() {
+          inProgress = false;
+        });
+        timer.cancel();
+      }
+    });
+  }
+
+  void _stopTimer() {
+    setState(() {
+      timeLeft = 0;
+      inProgress = false;
+    });
+  }
+
+  void _restartTimer() {
+    setState(() {
+      timeLeft = lastValue;
+    });
+    _startCountDown();
+  }
 
   @override
   Widget build(BuildContext context) {
-    return const Scaffold(
-        /// displays the "Pomodoro Timer" message at the top
-        appBar: CustomAppBar(title: "Pomodoro Timer"),
-        body: Text(
-          "Pomodoro timer coming soon...",
-          style: TextStyle(
-            color: Colors.black,
-            fontSize: 30.0,
+    return Scaffold(
+      appBar: const CustomAppBar(title: "Timer"),
+      body: Row(
+        mainAxisAlignment: MainAxisAlignment.center,
+        children: [
+          // minutes wheel
+          Container(
+            width: 70,
+            child: ListWheelScrollView.useDelegate(
```

Comment:
Does this work for all devices (android and ios)

Reply:
@lou-weng I can't speak to iOS, but it seems to be working fine for Android so I am going to merge